loader_hub/web/beautiful_soup_web/base.py

```diff
@@ -15,16 +15,67 @@ def _substack_reader(soup: Any) -> Tuple[str, Dict[str, Any]]:
     text = soup.select_one("div.available-content").getText()
     return text, extra_info
 
+def _readthedocs_reader(soup: Any,url: str) -> Tuple[str, Dict[str, Any]]:
+    """Extract text from a ReadTheDocs documentation site
 
-def _readthedocs_reader(soup: Any) -> Tuple[str, Dict[str, Any]]:
-    """Extract text from a ReadTheDocs documentation site"""
-    text = soup.find_all("main", {"id": "main-content"})[0].get_text()
-    return "\n".join([t for t in text.split("\n") if t]), {}
+    """
+    links = soup.find_all('a', {'class': 'reference internal'})
+    rtd_links = []
+    
+    for link in links:
+        rtd_links.append(link['href'])   
+    for i in range(len(rtd_links)):
+        if not rtd_links[i].startswith("http"):
+            rtd_links[i] = url + rtd_links[i]
+    texts = []
+    broken_docs = []
+    
+    for doc_link in rtd_links:
+        import requests
+        from bs4 import BeautifulSoup 
+        page_link = requests.get(doc_link)        
+        soup = BeautifulSoup(page_link.text, 'html.parser')
+        try:
+            text = soup.find(attrs={"role": "main"}).get_text()
+            
+        except IndexError:
+            text = None
+            broken_docs.append(doc_link)
+        if text:
+            texts.append("\n".join([t for t in text.split("\n") if t]))
+    return '\n'.join(texts), {}
+            
+def _readmedocs_reader(soup: Any,url: str) -> Tuple[str, Dict[str, Any]]:
+    """Extract text from a ReadMe documentation site"""
+    
+    links = soup.find_all("a")
+    docs_links = [link["href"] for link in links if "/docs/" in link["href"]]
+    docs_links = list(set(docs_links))
+    for i in range(len(docs_links)):
+        if not docs_links[i].startswith("http"):
+            docs_links[i] = url + docs_links[i]
+            
+    texts = []
+    for doc_link in docs_links[:5]:
+        import requests
+        from bs4 import BeautifulSoup 
+        page_link = requests.get(doc_link)
+        soup = BeautifulSoup(page_link.text, 'html.parser')
+        try:
+            text = soup.find_all("article", {"id": "content"})[0].get_text()
+        except IndexError:
+            text = None
+            raise IndexError("No content found")
+
+        if text:
+            texts.append("\n".join([t for t in text.split("\n") if t]))
+    return '\n'.join(texts), {}
 
 
 DEFAULT_WEBSITE_EXTRACTOR: Dict[str, Callable[[Any], Tuple[str, Dict[str, Any]]]] = {
```

Comment:
If `url` is now an arg, the type here is now `Dict[str, Callable[[Any, str], Tuple[str, Dict[str, Any]]]] `